client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -87,16 +88,40 @@
 
     /**
      * API to merge or split stream segments.
+     *
+     * @param stream Stream object.
+     * @param sealedSegments List of segments to be sealed.
+     * @param newKeyRanges Key ranges after scaling the stream.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the scaling was started or not.
+     */
+    CompletableFuture<Boolean> startScale(final Stream stream, final List<Integer> sealedSegments,
+                                           final Map<Double, Double> newKeyRanges);
+
+    /**
+     * API to merge or split stream segments.
      * 
      * @param stream Stream object.
      * @param sealedSegments List of segments to be sealed.
      * @param newKeyRanges Key ranges after scaling the stream.
+     * @param timeout timeout and throw exception if scale does not complete withing specified timeout.
```

Comment:
completableFuture will block for timebound fashion.. 
But what we do here is we make checkScaleStatus call in a loop. We want to do that only for limited number of times (determined by timeout supplied). 
If we dont do that, but only rely on completableFuture.get(timeout) then the caller will block for timebound manner, but the future's processing will keep on running forever until the scale completes. which doesnt serve any purpose.